### PR TITLE
[23416] Covert vaccine flow to hooks

### PR DESCRIPTION
**Pull request by LanceGundersen**
## Description
We want to update the way we use react-redux to the current recommended approach, which is to use the useSelector and useDispatch hooks.

Paths:
src/applications/vaos/covid-19-vaccine/components/PlanAheadPage.jsx
src/applications/vaos/covid-19-vaccine/components/ReceivedDoseScreenerPage.jsx
src/applications/vaos/covid-19-vaccine/components/ReviewPage.jsx
src/applications/vaos/covid-19-vaccine/components/SecondDosePage.jsx
src/applications/vaos/covid-19-vaccine/components/SelectDate1Page.jsx
src/applications/vaos/covid-19-vaccine/components/VAFacilityPage/index.jsx

## Acceptance criteria
- [ ] connect() is no longer used on page
- [ ] All actions that were in mapDispatchToProps are now wrapped in dispatch() when called
- [ ] All data in mapStateToProps is pulled in through useSelector hooks

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/covid-19-vaccine/index.jsx [hunk 1]
```diff
@@ -33,20 +32,22 @@ import {
...
+  const canUseVaccineFlow = useSelector(selectCanUseVaccineFlow);
+  const featureCovid19Vaccine = useSelector(selectFeatureCovid19Vaccine);
+  const directScheduleSettingsStatus = useSelector(
+    selectDirectScheduleSettingsStatus,
+  );
 
   useEffect(() => {
     if (directScheduleSettingsStatus === FETCH_STATUS.notStarted) {
-      fetchDirectScheduleSettings();
+      dispatch(fetchDirectScheduleSettings);
```
Comment (jbalboni) [inline src/applications/vaos/covid-19-vaccine/index.jsx:50]: Missing `()` at the end of `fetchDirectScheduleSettings`

Comment (LanceGundersen) [reply src/applications/vaos/covid-19-vaccine/index.jsx:50]: @jbalboni good catch but still didn't fix the default error. I am assuming when I pulled the connect it broke something I am not seeing

File: src/applications/vaos/covid-19-vaccine/index.jsx [hunk 2]
```diff
@@ -33,20 +32,22 @@ import {
...
 } from '../appointment-list/redux/selectors';
+import { fetchDirectScheduleSettings } from '../appointment-list/redux/actions';
 
-export function NewBookingSection({
-  canUseVaccineFlow,
-  featureCovid19Vaccine,
-  directScheduleSettingsStatus,
-  fetchDirectScheduleSettings,
-}) {
+export default function NewBookingSection() {
```
Comment (jbalboni) [inline src/applications/vaos/covid-19-vaccine/index.jsx:37]: This wasn't a default export before and the tests are importing the function assuming it's not the default.

File: src/applications/vaos/covid-19-vaccine/components/SecondDosePage.jsx [hunk 3]
```diff
@@ -1,23 +1,29 @@
...
-function SecondDosePage({
-  data,
-  facility,
-  pageChangeInProgress,
-  routeToNextAppointmentPage,
-  routeToPreviousAppointmentPage,
-}) {
+export default function SecondDosePage() {
+  const { data, facility } = useSelector(
+    state => getCovid19VaccineFormPageInfo(state, pageKey),
```
Comment (jbalboni) [inline src/applications/vaos/covid-19-vaccine/components/SecondDosePage.jsx:21]: This is a different selector from the one used in mapStateToProps in this component.

File: src/applications/vaos/covid-19-vaccine/index.jsx (hunk 2 above)
Comment (LanceGundersen) [reply src/applications/vaos/covid-19-vaccine/index.jsx]: Ok yeah I see what's wrong now.

File: src/applications/vaos/covid-19-vaccine/components/SecondDosePage.jsx (hunk 3 above)
Comment (LanceGundersen) [reply src/applications/vaos/covid-19-vaccine/components/SecondDosePage.jsx:21]: Gotcha, yes I see the problem now

**Review by jbalboni — COMMENTED** [review]
Looks like there are some unit test failures related to your changes, I can take a look once those are fixed.

File: src/applications/vaos/covid-19-vaccine/components/SecondDosePage.jsx [hunk 4]
```diff
@@ -1,23 +1,26 @@
...
-  pageChangeInProgress,
-  routeToNextAppointmentPage,
-  routeToPreviousAppointmentPage,
-}) {
+export default function SecondDosePage() {
+  const { data, facility } = useSelector(
+    state => getReviewPage(state, pageKey),
+    shallowEqual,
+  );
+  const pageChangeInProgress = useSelector(selectPageChangeInProgress);
```
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/covid-19-vaccine/components/SecondDosePage.jsx:21]: removed

File: src/applications/vaos/covid-19-vaccine/components/SelectDate1Page.jsx [hunk 5]
```diff
@@ -57,60 +61,63 @@ function validate({ dates, setValidationError }) {
...
+          moment()
+            .add(1, 'months')
+            .endOf('month')
+            .format('YYYY-MM-DD'),
+          true,
+        ),
+      );
+      document.title = `${pageTitle} | Veterans Affairs`;
+    },
+    [dispatch],
```
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/covid-19-vaccine/components/SelectDate1Page.jsx]: removed

**Review by jbalboni — APPROVED** [review]

**Pull request merged**